spring-vault/src/test/java/com/baeldung/springvault/VaultIntegrationManualTest.java

```diff
@@ -1,90 +1,76 @@
+import org.springframework.vault.core.VaultTemplate;
+import org.springframework.vault.repository.configuration.EnableVaultRepositories;
 
 /**
  * These tests are requiring the {@code vault} command to be installed and available in the executing
  * platform. So, if you intend to run them in your environment, the please install the vault and then
  * run the ignored tests.
  */
-@RunWith(SpringRunner.class)
+@RunWith(SpringJUnit4ClassRunner.class)
 @SpringBootTest(classes = CredentialsService.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 @ContextConfiguration(classes = VaultTestConfiguration.class, loader = AnnotationConfigContextLoader.class)
-@FixMethodOrder(MethodSorters.NAME_ASCENDING)
 @DirtiesContext(classMode = ClassMode.AFTER_CLASS)
-public class VaultIntegrationTest {
-
+@EnableVaultRepositories
+public class VaultIntegrationManualTest {
     @Autowired
-    private CredentialsService credentialsService;
-
-    @MockBean
     private CredentialsRepository credentialsRepository;
 
-    /**
-     * Test to secure credentials.
-     *
-     * @throws URISyntaxException
-     */
-    @Test
-    @Ignore
-    public void givenCredentials_whenSecureCredentials_thenCredentialsSecured() throws URISyntaxException {
-        try {
-            // Given
-            Credentials credentials = new Credentials("username", "password");
-
-            // When
-            credentialsService.secureCredentials(credentials);
+    @Autowired
+    private VaultTemplate vaultTemplate;
 
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+    private CredentialsService credentialsService;
 
+    @Before
+    public void setup() {
+        this.credentialsService = new CredentialsService(vaultTemplate, credentialsRepository);
     }
 
     /**
-     * Test to access credentials
+     * Test to secure credentials.
      *
      * @throws URISyntaxException
      */
     @Test
-    @Ignore
-    public void whenAccessCredentials_thenCredentialsRetrieved() throws URISyntaxException {
-
+    public void givenCredentials_whenSecureCredentials_thenCredentialsSecuredSuccessfully() throws URISyntaxException {
         // Given
-        Credentials credentials = credentialsService.accessCredentials();
-
+        Credentials actualCredentials = new Credentials("username", "password");
+        // When
+        credentialsService.secureCredentials(actualCredentials);
         // Then
-        assertNotNull(credentials);
-        assertEquals("username", credentials.getUsername());
-        assertEquals("password", credentials.getPassword());
+        Credentials storedCredentials = credentialsService.accessCredentials(actualCredentials.getUsername());
+        Assertions.assertNotNull(storedCredentials);
+        Assertions.assertEquals(actualCredentials.getUsername(), storedCredentials.getUsername());
+        Assertions.assertEquals(actualCredentials.getPassword(), storedCredentials.getPassword());
```

Comment:
The order of arguments should be swapped for these assertEquals calls -- they should be assertEquals(expected, actual).

Reply:
Thanks, @KevinGilmore, for catching this. I've fixed the issue.